### ISSUE TITLE
Compilation issues GCC 6.1.X

### ISSUE THREAD

**Issue by figue** [arnestig/quickrdp#2]
Hello, I'm trying to build Quickrdp in my Archlinux machine trought this package in AUR, but I'm having GCC issues (I think), you can check all comments here:
[https://aur.archlinux.org/packages/quickrdp/?comments=all](url)

Compilation output:

```
In file included from src/CommandDatabase.h:28:0,
from src/CommandDialog.h:25,
from src/CommandDialog.cpp:22:
src/RDPDatabase.h:39:60: error: 'ConnectionType::getConnectionTypeName' declared as an 'inline' variable
inline wxString getConnectionTypeName( ConnectionType::ConnectionType connectionType )
^~~~~~~~~~~~~~
src/RDPDatabase.h:39:44: error: 'ConnectionType' is not a member of 'ConnectionType::ConnectionType'
inline wxString getConnectionTypeName( ConnectionType::ConnectionType connectionType )
^~~~~~~~~~~~~~
src/RDPDatabase.h:40:5: error: expected ',' or ';' before '{' token
{
^
In file included from /usr/include/wx-3.0/wx/gdicmn.h:23:0,
from /usr/include/wx-3.0/wx/event.h:20,
from /usr/include/wx-3.0/wx/window.h:18,
from /usr/include/wx-3.0/wx/sizer.h:16,
from src/ExampleDialog.h:26,
from src/CommandDialog.h:26,
from src/CommandDialog.cpp:22:
/usr/include/wx-3.0/wx/math.h:116:39: error: expected '}' before end of line
wxGCC_WARNING_SUPPRESS(float-equal)
^
/usr/include/wx-3.0/wx/math.h:116:39: error: expected declaration before end of line
make: *** [Makefile:45: obj/CommandDialog.o] Error 1
make: *** Waiting for unfinished jobs....
```

I don't have the knowledge to help with the patches. I would very much appreciate it if you could fix it.

Thank you

**Comment by arnestig** [comment]
Hi, i haven't tested gcc > 4.9 for compiling quickrdp or any other wxWidget applications.

Looks like you both have some issues with declarations in RDPDatabase.h and also in wxWidgets math.h

Even if I can solve the problem for RDPDatabase.h I think you will still have some issues with wxWidget.

Will see if I can get hold of a copy of gcc 6.1 and solve the issue in RDPDatabase.h however :)

**Comment by arnestig** [comment]
I could reproduce this and solve it with the fix in 675c94a.

Hope it works for you
